tests/console/btrfsmaintenance.pm

```diff
@@ -0,0 +1,85 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Check btrfsmainenance for functionality
+# - run btrfs-balance.sh and btrfs-scrub.sh
+# - Check if btrfsmaintenance-refresh.service is present and started properly
+# - Check if btrfs-scrub is scheduled
+# - Check if btrfs-balance is scheduled
+# - Checks for btrfs-related cron jobs after uninstalling btrfsmaintenance
+# Maintainer: Felix Niederwanger <felix.niederwanger@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use registration qw(cleanup_registration register_product add_suseconnect_product get_addon_fullname remove_suseconnect_product);
+use version_utils "is_sle";
+
+my $btrfsmaintenance_removed = 0;
+
+sub btrfs_service_check {
+    my $service = $_[0];
+    if (script_run("systemctl status btrfsmaintenance-refresh.service | grep -E -v 'uninstall|none' | grep $service") == 0) {
+        record_info("btrfs-$service", "btrfs-$service enabled via btrfsmaintenance-refresh.service");
+        return 0;
+    }
```

Comment:
this looks like you want something like https://progress.opensuse.org/issues/45410 :)

You could also try to use https://perldoc.perl.org/Test/More.html directly here instead of doing the "script_run, record_info, return 0" dance repeatedly :)

Reply:
Hmmm, in this test it is a cascade that requires checking three conditionals in a particular order to be checked. Since only three checks are made I would really like to avoid including an additional dependency here.

Yet, as a compromise I added a comment that should increase the understanding of what happens and went for the `elsif` block, so it becomes clearer that it's a cascading if block series - I hope this is in favor of resolving the issues you pointed out :-)

Reply:
I guess you could just use `assert_script_run` with a `fail_message`. E.g.

```
assert_script_run("systemctl status btrfsmaintenance-refresh.service | grep -E -v 'uninstall|none' | grep $service",
	fail_message => "btrfs-$service not enabled via btrfsmaintenance-refresh.service");
```

Reply:
Would be an option but the idea was to record the method that is in the end used instead of the ones that are not used. All of the three methods are valid, we're grepping for the actual one that is used.

Reply:
I did not get that but it's ok to have your choice :)

Reply:
I appreciate your input Oliver, thanks :slightly_smiling_face: